src/event-handlers.test.js

```diff
@@ -168,3 +168,48 @@ describe('handleCowSelect', () => {
     expect(component.instance().selectCow).toHaveBeenCalledWith({ id: 'abc' })
   })
 })
+
+describe('handleShowHomeScreenChange', () => {
+
+  beforeEach(() => {
+    jest.spyOn(component.instance(), 'setState').mockImplementation()
+    jest.spyOn(component.instance(), 'focusNextView').mockImplementation()
+  })
+
+  test('change show home screen setting to False and navigate to the next view', () => {
+    component.state().stageFocus = stageFocusType.HOME
+    component.state().showHomeScreen = true
+
+    handlers().handleShowHomeScreenChange(null, false)
+    expect(component.instance().setState).toHaveBeenCalled()
+    // expect(component.showHomeScreen).toBeFalse()
+    expect(component.instance().focusNextView).toHaveBeenCalled()
+  })
+
+  test('change show home screen setting to False and do not navigate', () => {
+    component.state().stageFocus = stageFocusType.SHOP
+    component.state().showHomeScreen = true
+
+    handlers().handleShowHomeScreenChange(null, false)
+    expect(component.instance().setState).toHaveBeenCalled()
+    expect(component.instance().focusNextView).not.toHaveBeenCalled()
+  })
+
+  test('change show home screen setting to True and does not navigate', () => {
+    component.state().stageFocus = stageFocusType.HOME
+    component.state().showHomeScreen = false
+
+    handlers().handleShowHomeScreenChange(null, true)
+    expect(component.instance().setState).toHaveBeenCalled()
+    expect(component.instance().focusNextView).not.toHaveBeenCalled()
+  })
+
+  test('change show home screen setting to True and does not navigate', () => {
+    component.state().stageFocus = stageFocusType.SHOP
+    component.state().showHomeScreen = false
+
+    handlers().handleShowHomeScreenChange(null, true)
+    expect(component.instance().setState).toHaveBeenCalled()
+    expect(component.instance().focusNextView).not.toHaveBeenCalled()
+  })
+})
```

Comment:
Sorry for just dropping a big diff here, but I spent some time with this and this and this change seems to get things going:

```diff
diff --git a/src/event-handlers.test.js b/src/event-handlers.test.js
index 1d97569..b99b98a 100644
--- a/src/event-handlers.test.js
+++ b/src/event-handlers.test.js
@@ -170,46 +170,48 @@ describe('handleCowSelect', () => {
 })
 
 describe('handleShowHomeScreenChange', () => {
-
   beforeEach(() => {
-    jest.spyOn(component.instance(), 'setState').mockImplementation()
     jest.spyOn(component.instance(), 'focusNextView').mockImplementation()
   })
 
   test('change show home screen setting to False and navigate to the next view', () => {
-    component.state().stageFocus = stageFocusType.HOME
-    component.state().showHomeScreen = true
+    component.setState({
+      stageFocus: stageFocusType.HOME,
+      showHomeScreen: true,
+    })
 
     handlers().handleShowHomeScreenChange(null, false)
-    expect(component.instance().setState).toHaveBeenCalled()
-    // expect(component.showHomeScreen).toBeFalse()
+    expect(component.state().showHomeScreen).toBeFalse()
     expect(component.instance().focusNextView).toHaveBeenCalled()
   })
 
   test('change show home screen setting to False and do not navigate', () => {
-    component.state().stageFocus = stageFocusType.SHOP
-    component.state().showHomeScreen = true
+    component.setState({
+      stageFocus: stageFocusType.SHOP,
+      showHomeScreen: true,
+    })
 
     handlers().handleShowHomeScreenChange(null, false)
-    expect(component.instance().setState).toHaveBeenCalled()
     expect(component.instance().focusNextView).not.toHaveBeenCalled()
   })
 
   test('change show home screen setting to True and does not navigate', () => {
-    component.state().stageFocus = stageFocusType.HOME
-    component.state().showHomeScreen = false
+    component.setState({
+      stageFocus: stageFocusType.HOME,
+      showHomeScreen: false,
+    })
 
     handlers().handleShowHomeScreenChange(null, true)
-    expect(component.instance().setState).toHaveBeenCalled()
     expect(component.instance().focusNextView).not.toHaveBeenCalled()
   })
 
   test('change show home screen setting to True and does not navigate', () => {
-    component.state().stageFocus = stageFocusType.SHOP
-    component.state().showHomeScreen = false
+    component.setState({
+      stageFocus: stageFocusType.SHOP,
+      showHomeScreen: false,
+    })
 
     handlers().handleShowHomeScreenChange(null, true)
-    expect(component.instance().setState).toHaveBeenCalled()
     expect(component.instance().focusNextView).not.toHaveBeenCalled()
   })
 })
```

Basically, I think we need to drop `jest.spyOn(component.instance(), 'setState').mockImplementation()` in the `beforeEach` (because we want to use it as-is), use `setState` to set up desired starting states to test, and drop the `expect(component.instance().setState).toHaveBeenCalled()` assertions.

I think this could use some additional `expect(component.state().showHomeScreen).toBeFalse()`/`toBeTrue()`'s as necessary to round it out, but the diff above should help to get things into a better place to work from at least. Let me know if you have additional questions about this!

Reply:
That cleared it all up, thanks! :)